### ISSUE TITLE
How to uniquely identify a project?

### ISSUE THREAD

**Issue by sckott** [njahn82/ropenaire#8]
@njahn82 @alessiabardi  

We assume that projectID's are not unique, correct? In that case can we uniquely identify a project with funder ID/name + project ID?

Did i get the question right @pkraker ?

**Comment by alessiabardi** [comment]
Yes, different funders might use the same project ID. Using the pair funder ID/name + project ID you ensures you are uniquely identifying a project

**Comment by sckott** [comment]
@pkraker see ^^

**Comment by sckott** [comment]
thanks @alessiabardi

**Comment by pkraker** [comment]
Thanks! @alessiabardi Following the [documentation](api.openaire.eu/api.html), I assume that this would be parameters `funder` and `projectID`, correct? The `datasets` endpoint, however, does not have a parameter `projectID`. How can this be achieved at this endpoint? In addition, you have listed only 7 shortcodes for funders (`WT | EC | ARC | ANDS | NSF | FCT | NHMRC`), what would be the others?

**Comment by alessiabardi** [comment]
The dataset endpoint does have the projectID parameter (e.g. http://api.openaire.eu/search/datasets?projectID=645322). For the funder short code, which list are you referring to? In the API documentation more funder codes are listed: http://api.openaire.eu/bulk-projects.html . Unfortunately the list is not updated automatically and cannot be programmatically retrieved. This is something we plan to add in the not-so-far future.

**Comment by pkraker** [comment]
@alessiabardi Thanks, that helps a lot. One final question that @tarrow and I came across today: in [the bulk projects documentation](http://api.openaire.eu/bulk-projects.html), `FP7` and `H2020` are listed as funder codes, but when you [use them with the API](http://api.openaire.eu/search/publications?funder=FP7&size=10), no projects are returned. Are we correct in assuming that for FP7 and H2020, we should simply use `EC` as funder code?

**Comment by alessiabardi** [comment]
Correct. We had to keep FP7 and H2020 codes as "funders" in the bulk projects API for legacy code in the ePrints/DSpace plugins that are using those APIs. For the search API, in fact, we named parameters in the correct way: FP7 and H2020 are _funding streams_ of the funder EC. If you are interested on EC project, then use _funder=EC_, if you are interested in FP7 projects, then: http://api.openaire.eu/search/publications?fundingStream=FP7&size=10.
